### DISCUSSION TITLE
There seems to be no examples of importing JavaScript. Can you tell me how to import JavaScript into the Leptos project

### DISCUSSION THREAD

**Discussion by lhjok** [leptos-rs/leptos#1471] [General]
There seems to be no examples of importing JavaScript. Can you tell me how to import JavaScript into the Leptos project

**Comment by stewartadam** [comment]
You would likely need to put the JavaScript file under the asset folder then include it in the HTML output by Leptos. In your `App` component function you would add a `<script>` tag. For example:

```
#[component]
pub fn App() -> impl IntoView {
    // Provides context that manages stylesheets, titles, meta tags, etc.
    provide_meta_context();

    view! {
        // injects a stylesheet into the document <head>
        // id=leptos means cargo-leptos will hot-reload this stylesheet
        <Stylesheet id="leptos" href="/pkg/style.css"/>

        // sets the document title
        <Title text="Welcome to Leptos"/>

        // content for this welcome page
        <Router>
            <main>
                <Routes>
                    <Route path="" view=|| view! { <HomePage/> }/>
                </Routes>
            </main>
        </Router>
        <script src="/js/script.js"></script>
    }
}
```

Assuming that you have a `js` folder and `script.js` within it created in your asset directory.

**Comment by gbj** [comment]
We had the same discussion on Discord. For anyone future readers: wasm-bindgen is how you import JavaScript into any Rust/WASM project so that you can call it from Rust https://rustwasm.github.io/wasm-bindgen/examples/import-js.html

**Reply by qrilka** [reply]
@gbj I see this question in multiple threads here, probably it could be worth to have a small example how to do this?

**Reply by gbj** [reply]
@qrilka Sounds good. Would you be interested in contributing one?

**Reply by qrilka** [reply]
I'll try to find some time for that

**Reply by qidian99** [reply]
any updates?

In my case I'd call a js callback on state update, e.g., 
```js

        <script>
            hljs.highlightAll();
        </script>
```

**Reply by qrilka** [reply]
could you share a small js example maybe? From the snippet it's not clear what library you use

**Reply by gbj** [reply]
@qidian99 I would strongly recommend reading [the `wasm-bindgen` docs linked above](https://rustwasm.github.io/wasm-bindgen/examples/import-js.html) on how to use JS functions from Rust/WASM. I've [added a section to the Leptos book](https://book.leptos.dev/web_sys.html#using-js-libraries-with-wasm-bindgen) but there's nothing really Leptos-specific here at all.
